### PR TITLE
DOC: improve lax.dot_general documentation

### LINKED ISSUE #14670: jax.lax.dot_general has minimal documentation
Please:

- [x] Check for duplicate requests.
- [x] Describe your goal, and if possible provide a code snippet with a motivating example.

----------

The docs for [`jax.lax.dot_general`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.dot_general.html) are unclear. In particular, the `dimension_numbers` argument isn't clear to me.

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #14670

### CODE REVIEW

File: jax/_src/lax/lax.py [hunk 1]
```diff
@@ -747,18 +747,25 @@ def dot(lhs: Array, rhs: Array, precision: PrecisionLike = None,
...
+  `DotGeneral  <https://www.tensorflow.org/xla/operation_semantics#dotgeneral>`_
+  operator documentation.
+
   Args:
     lhs: an array
     rhs: an array
-    dimension_numbers: a tuple of tuples of the form
-      `((lhs_contracting_dims, rhs_contracting_dims),
-      (lhs_batch_dims, rhs_batch_dims))`
+    dimension_numbers: a tuple of tuples of sequences of the form
```
Comment (mattjj) [inline jax/_src/lax/lax.py:767]:
```suggestion
    dimension_numbers: a tuple of tuples of sequences of ints, of the form
```

File: jax/_src/lax/lax.py [hunk 2]
```diff
@@ -768,7 +775,9 @@ def dot_general(lhs: ArrayLike, rhs: ArrayLike, dimension_numbers: DotDimensionN
       accumulate results to and return a result with that datatype.
 
   Returns:
-    An array containing the result.
+    An array whose first dimensions are the batch dimensions, followed by the the ``lhs``
```
Comment (mattjj) [inline jax/_src/lax/lax.py:778]:
```suggestion
    An array whose first dimensions are the (shared) batch dimensions, followed by the the ``lhs``
```

**Review by mattjj — APPROVED** [review]
LGTM! Optional suggestions. Thanks for fixing this.

File: jax/_src/lax/lax.py (hunk 1 above)
Comment (jakevdp) [reply jax/_src/lax/lax.py]: Done!

File: jax/_src/lax/lax.py (hunk 2 above)
Comment (jakevdp) [reply jax/_src/lax/lax.py]: Done

**Pull request merged**